src/NHibernate/Cache/CacheFactory.cs

```diff
@@ -67,16 +67,14 @@ public static class CacheFactory
 						"cache usage attribute should be read-write, read-only, nonstrict-read-write, or transactional");
 			}
 
-			ICache impl;
 			try
 			{
-				impl = settings.CacheProvider.BuildCache(name, properties);
+				ccs.Cache = settings.CacheProvider.BuildCache(name, properties);
 			}
 			catch (CacheException e)
```

Comment:
Anyway I do not know of any cache implementation throwing this exception... Those from NHibernate-Caches never thrown it.